src/openapi/transport/trasportBase.ts

```diff
@@ -0,0 +1,71 @@
+import type { Methods, TransportCoreOptions } from './types';
+
+type MethodReturn = (
+    servicePath?: string,
+    urlTemplate?: string,
+    templateArgs?: Record<string, string | number> | null,
+    options?: TransportCoreOptions,
+) => Promise<
+    | {
+          response?: string | Blob;
+          status: number;
+          headers: {
+              get: (key: string) => string;
+          };
+          size: number;
+          url: string;
+          responseType?: string;
+          isNetworkError?: boolean;
+      }
+    | void
+    | unknown
+>;
+
+export interface ITransport {
+    prepareFunction: (arg0: Methods) => MethodReturn;
+    dispose: () => void;
+    get: MethodReturn;
+    post: MethodReturn;
+    put: MethodReturn;
+    delete: MethodReturn;
+    patch: MethodReturn;
+    head: MethodReturn;
+    options: MethodReturn;
+}
+
+abstract class TransportBase implements ITransport {
+    abstract dispose(): void;
+
+    abstract prepareFunction(

```

Comment:
What about naming this `prepareHttpMethod` or `prepareTransportMethod`? This one is very general.